src/controllers/accommodationController.js

```diff
@@ -190,4 +202,35 @@ export default class AccommodationController {
     });
   }
 
+  static async viewSpecificAccommodation(req, res) {
+    const { role } = req.user.payload;
+    const { slug } = req.params;
+    const { Op } = sequelize;
+
+    if (role === 2) {
+      const accommodation = await models.accommodations.findOne({
```

Comment:
Make this a helper function to avoid repeatition. You can use something like this
```
const getAccomodation =(dataObj)=>{
cpnst x = models.accommodations.findOne({
        where: dataObj,
        attributes: { exclude: ['locationId', 'owner'] },
        include: [
          { model: models.users, as: 'ownerUser', attributes: ['id', 'email', 'phone'] },
          { model: models.locations, as: 'accommodationLocation', attributes: ['id', 'name'] },
        ]
      });
if(!x){error}
return x
}
```